factory/apiResolverFactory.go

```diff
@@ -539,6 +540,11 @@ func createAPIBlockProcessorArgs(args *ApiResolverArgs, apiTransactionHandler ex
 		return nil, err
 	}
 
+	receiptsRepository, err := createReceiptsRepository(args)
```

Comment:
you can create the receipts repository in the process components, and have it accessible from it

args.ProcessComponents.ReceiptsRepository()

Reply:
Fixed.